### PR TITLE
Add an AWS-managed Elasticsearch and configure Vector to ship data to it via elasticsearch sink

### LINKED ISSUE #4: Ship logs to AWS-managed Elasticsearch

### PR DESCRIPTION
**Pull request by MOZGIII**
Closes #4.

### CODE REVIEW

**Comment by MOZGIII** [comment]
I'm getting errors like this one at Vector deployed via this PR:

```
Dec 22 06:47:22.301 ERROR sink{component_kind="sink" component_name=elasticsearch component_type=elasticsearch}:request{request_id=35}: vector::sinks::util::sink: Response wasn't successful. response=Response { status: 403, version: HTTP/1.1, headers: {"date": "Tue, 22 Dec 2020 06:47:22 GMT", "content-type": "application/json", "content-length": "187", "connection": "keep-alive", "x-amzn-requestid": "4fccc244-8599-44e9-93f9-80e48ce3e017", "access-control-allow-origin": "*"}, body: b"{\"Message\":\"User: arn:aws:sts::071959437513:assumed-role/eksctl-k8s-dev-env-nodegroup-ng-1-NodeInstanceRole-19MM262WR0FHX/i-069c5c6485f9f830b is not authorized to perform: es:ESHttpPost\"}" }
```

I guess the AWS identity provider catches up properly (unless it's a Node-level AWS access), but then it means there is a problem with AWS ES access configuration.

**Comment by MOZGIII** [comment]
Here's what tests I've conducted:

```
$ aws-vault exec vector -- kubectl run -n vector --image amazon/aws-cli -i --rm access-test -- sts get-caller-identity
If you don't see a command prompt, try pressing enter.
Error attaching, falling back to logs: unable to upgrade connection: container access-test not found in pod access-test_vector
{
    "UserId": "AROARBQI54DE32TGWD7OT:i-069c5c6485f9f830b",
    "Account": "071959437513",
    "Arn": "arn:aws:sts::071959437513:assumed-role/eksctl-k8s-dev-env-nodegroup-ng-1-NodeInstanceRole-19MM262WR0FHX/i-069c5c6485f9f830b"
}
pod "access-test" deleted

$ aws-vault exec vector -- kubectl run -n vector --image amazon/aws-cli -i --rm access-test --serviceaccount vector-aggregator -- sts get-caller-identity
If you don't see a command prompt, try pressing enter.
Error attaching, falling back to logs: unable to upgrade connection: container access-test not found in pod access-test_vector

An error occurred (AccessDenied) when calling the AssumeRoleWithWebIdentity operation: Not authorized to perform sts:AssumeRoleWithWebIdentity
pod "access-test" deleted
```

So, what this concludes is that `vector-aggregator` service account has it's own role, which is very restricted - it can't even request it's own identity (which is fine).
Another find is that `arn:aws:sts::071959437513:assumed-role/eksctl-k8s-dev-env-nodegroup-ng-1-NodeInstanceRole-19MM262WR0FHX/i-069c5c6485f9f830b` is actully a Node-level access role, not the service-account-level role. Vector should *not* be using that.

**Comment by MOZGIII** [comment]
False alarm!

I was wrong, the `elasticsearch` sink does work properly.

After fixing the service account role it works as intended.

What could be done better is error reporting when we get the `Not authorized to perform sts:AssumeRoleWithWebIdentity`.

File: terraform/aws/es/variables.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+variable "domain_name" {
+  type        = string
+  default     = "k8s-dev-env"
+  description = "AWS ElasticSearch Domain name"
+}
+
+variable "access_policies" {
```
Comment (jszwedko) [inline terraform/aws/es/variables.tf:7]: Does this need to be dynamic? I'm curious what you are passing in as the value for this.

Comment (MOZGIII) [reply terraform/aws/es/variables.tf:7]: It's just a small convenience, here's how I use it:

https://github.com/timberio/vector-k8s-dev-env/blob/98dd367da906cbe38bc025d4b1d643304f6bbc58/terraform/aws/main.tf#L76-L80

Comment (jszwedko) [reply terraform/aws/es/variables.tf:7]: Aha :+1:

**Review by jszwedko — APPROVED** [review]
Looks good to me!

**Pull request merged**